traffic_ops/traffic_ops_golang/cdnnotification/cdnnotifications.go

```diff
@@ -0,0 +1,191 @@
+`
+
+const deleteQuery = `
+DELETE FROM cdn_notification
+WHERE cdn_notification.cdn = $1
+RETURNING cdn_notification.cdn,
+          cdn_notification.last_updated,
+          cdn_notification.user,
+          cdn_notification.notification
+`
+
+// Read is the handler for GET requests to /cdn_notifications.
+func Read(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	cdnNotifications := []tc.CDNNotification{}
+
+	queryParamsToQueryCols := map[string]dbhelpers.WhereColumnInfo{
+		"cdn":  dbhelpers.WhereColumnInfo{"cdn.name", nil},
+		"user": dbhelpers.WhereColumnInfo{"tm_user.username", nil},
+	}
+
+	where, orderBy, pagination, queryValues, errs := dbhelpers.BuildWhereAndOrderByAndPagination(inf.Params, queryParamsToQueryCols)
+	if len(errs) > 0 {
+		sysErr = util.JoinErrs(errs)
+		errCode = http.StatusBadRequest
+		api.HandleErr(w, r, tx, errCode, nil, sysErr)
+		return
+	}
+
+	query := readQuery + where + orderBy + pagination
+	rows, err := inf.Tx.NamedQuery(query, queryValues)
+	if err != nil {
+		userErr, sysErr, errCode = api.ParseDBError(err)
+		if sysErr != nil {
+			sysErr = fmt.Errorf("notification read query: %v", sysErr)
+		}
+
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var n tc.CDNNotification
+		if err = rows.Scan(&n.CDN, &n.LastUpdated, &n.User, &n.Notification); err != nil {
+			api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, errors.New("scanning cdn notifications: "+err.Error()))
+			return
+		}
+		cdnNotifications = append(cdnNotifications, n)
+	}
+
+	api.WriteResp(w, r, cdnNotifications)
+}
+
+// Create is the handler for POST requests to /cdn_notifications.
+func Create(w http.ResponseWriter, r *http.Request) {
+	inf, sysErr, userErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if sysErr != nil || userErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	var req tc.CDNNotificationRequest
+	if userErr = api.Parse(r.Body, tx, &req); userErr != nil {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, userErr, nil)
+		return
+	}
+
+	var resp tc.CDNNotification
+	err := tx.QueryRow(insertQuery, req.CDN, inf.User.UserName, req.Notification).Scan(&resp.CDN, &resp.LastUpdated, &resp.User, &resp.Notification)
+	if err != nil {
+		userErr, sysErr, errCode = api.ParseDBError(err)
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+
+	changeLogMsg := fmt.Sprintf("CDN_NOTIFICATION: %s, CDN: %s, ACTION: Created", *resp.Notification, resp.CDN)
+	api.CreateChangeLogRawTx(api.ApiChange, changeLogMsg, inf.User, tx)
+
+	alertMsg := fmt.Sprintf("CDN notification created [ User = %s ] for CDN: %s", resp.User, resp.CDN)
+	api.WriteRespAlertObj(w, r, tc.SuccessLevel, alertMsg, resp)
```

Comment:
creation should use a `201 Created` response code and ideally set a `Location` header with to a request path where the new object representation can be fetched.

Reply:
where can i find an example of that?

Reply:
`traffic_ops/traffic_ops_golang/acme/acme_account.go`